hack/update-vendor.sh

```diff
@@ -30,13 +30,3 @@ TMP_DIR="${TMP_DIR:-$(mktemp -d /tmp/update-vendor.XXXX)}"
 trap "rm -rf $TMP_DIR" EXIT
 
 go mod vendor
-
-# Ensure sort order doesn't depend on locale
-export LANG=C
-export LC_ALL=C
-
-# sort recorded packages for a given vendored dependency in modules.txt.
-# `go mod vendor` outputs in imported order, which means slight go changes (or different platforms) can result in a differently ordered modules.txt.
-# scan                 | prefix comment lines with the module name       | sort field 1  | strip leading text on comment lines
-awk '{if($1=="#") print $2 " " $0; else print}' < vendor/modules.txt | sort -k1,1 -s | sed 's/.*#/#/' > "${TMP_DIR}/modules.txt.tmp"
-mv "${TMP_DIR}/modules.txt.tmp" vendor/modules.txt
```

Comment:
This change deserves a little explanation in case there's a better way to handle it.
If I ran `hack/update-vendor.sh && make verify` it failed with errors about inconsistent vendoring:
```
...
        k8s.io/utils@v0.0.0-20210707171843-4b05e18ac7d9: is explicitly required in go.mod, b
ut not marked as explicit in vendor/modules.txt
        sigs.k8s.io/sig-storage-lib-external-provisioner/v6@v6.3.0: is explicitly required i
n go.mod, but not marked as explicit in vendor/modules.txt

        To ignore the vendor directory, use -mod=readonly or -mod=mod.
        To sync the vendor directory, run:
                go mod vendor
make: *** [Makefile:50: verify] Error 1
```
If I ran `go mod vendor && make verify` it printed the diff and still failed:
```
...
< # k8s.io/pod-security-admission => k8s.io/pod-security-admission v0.22.1
< # k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.22.1
< # k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.22.1
< # k8s.io/sample-controller => k8s.io/sample-controller v0.22.1
make: *** [Makefile:50: verify] Error 1
```
Removing this sorting is the only way I see for `make verify` to pass. Perhaps this wasn't an issue with previous versions of go.